### PR TITLE
Get me logic with some tests refactoring

### PR DESCRIPTION
**Pull request by Nyks06**
Created the logic and tests for the get me (which get the currently connected user)

### CODE REVIEW

File: tests/helpers/users.js [hunk 1]
```diff
@@ -2,19 +2,17 @@ const request = require("supertest");
 
 const { generateUser } = require("../fixtures/users");
 const { app } = require("../../http/server");
+const { EXPECTATION_FAILED } = require("http-status-codes");
 
-const createUserAndLogin = async() => {
-	user = generateUser("session-create-generate-user");
-    responseCreate = await request(app.callback()).post("/users").send(user);
+const createUserAndLogin = async(seed = "generate-user") => {
```
Comment (topmrscode) [inline tests/helpers/users.js:7]: its better to turn it into a dictionary {seed:"users-get"}. Moreover pay attention to the name of the seed that you use in each tests in order not to mix between each file. For example for this test you can use "users-get" 😄

File: tests/helpers/users.js [hunk 2]
```diff
@@ -2,19 +2,17 @@ const request = require("supertest");
...
 
 	const responseLogin = await request(app.callback())
 	.post("/sessions")
 	.send(user);
 
-	return {
-		user: responseLogin.body.data.user,
-		token: responseLogin.body.data.token
-	}
+	return responseLogin;
```
Comment (topmrscode) [inline tests/helpers/users.js:15]: no its necessary to return all the responseLogin just user and token

File: tests/sessions/delete.test.js [hunk 3]
```diff
@@ -6,24 +6,36 @@ const { setup, app } = require("../../http/server");
...
     mongoServer = new MongoMemoryServer();
     const mongoUri = await mongoServer.getUri();
     await mongoose.connect(mongoUri);
     await setup();
 
-    resp = await createUserAndLogin()
-    token = resp.token
-    user = resp.user
-    userID = resp.user._id
+    const resp = await createUserAndLogin(seed = "delete-session");
```
Comment (topmrscode) [inline tests/sessions/delete.test.js:26]: change "resp" to something that allow us to know what this function return, example user taking into account the previous comments :)
name of the seed "sessions-delete"

File: tests/sessions/delete.test.js [hunk 4]
```diff
@@ -6,24 +6,36 @@ const { setup, app } = require("../../http/server");
...
+    expect(resp.status).toEqual(HttpStatus.OK)
+    expect(resp.body.status).toEqual({
+      code: HttpStatus.OK,
+      error: false,
+      identifier: IDEN_OK,
+      reason: REASON_OK,
+    })
+    expect(resp.body.data).not.toBe(null)
+
```
Comment (topmrscode) [inline tests/sessions/delete.test.js:35]: no its neseccary to verify this, its already done in login tests

File: tests/users/get-by-id.test.js [hunk 5]
```diff
@@ -32,18 +28,30 @@ describe("Get user by ID", () => {
...
+    token = resp.body.data.token;
+
+    const secondResp = await createUserAndLogin(seed = "get-by-id-snd");
+    expect(secondResp.status).toEqual(HttpStatus.OK)
+    expect(secondResp.body.status).toEqual({
+      code: HttpStatus.OK,
+      error: false,
+      identifier: IDEN_OK,
+      reason: REASON_OK,
+    })
```
Comment (topmrscode) [inline tests/users/get-by-id.test.js:51]: see previous comments about expect functions

File: tests/users/get-by-id.test.js [hunk 6]
```diff
@@ -78,7 +86,7 @@ describe("Get user by ID", () => {
 
     it("Should not be able to get an user if is not connected", async (done) => {
       const responseGet = await request(app.callback()).get(
-        "/users/" + createdUser._id
+        "/users/" + secondUser._id
       );
```
Comment (topmrscode) [inline tests/users/get-by-id.test.js:90]: you can just no add the .set not its necessary to use the secondUser

File: tests/helpers/users.js (hunk 2 above)
Comment (Nyks06) [reply tests/helpers/users.js:15]: I return all of the result in order, in the tests, to be able to check that the HTTP query worked well.
The problem is that here, I don't know if the response contains a data.user and data.token, wdyt ?

File: tests/helpers/users.js (hunk 1 above)
Comment (Nyks06) [reply tests/helpers/users.js:7]: Agree for the seed name 🙄  - on the other side not sure for the dict thing, why wouldn't we prefer to have a field ?

File: tests/sessions/delete.test.js (hunk 4 above)
Comment (Nyks06) [reply tests/sessions/delete.test.js:35]: It is not tested before, that's why I just want to confirm that the query has been done successfully.

**Pull request merged**